src/game/Strategic/Strategic_Status.h

```diff
@@ -99,7 +99,10 @@ struct STRATEGIC_STATUS
 	UINT8			ubNumNewSectorsVisitedToday;
 	UINT8			ubNumberOfDaysOfInactivity;
 
-	INT8 bPadding[70]; // XXX HACK000B
+	// For all weapons with an item index >= MAX_WEAPONS this array stores
+	// their "already dropped" status as a single bit. The size of this array
+	// must be exactly 70 because the struct size must be 192 bytes.
+	std::array<uint8_t, 70> additionalWeaponsAlreadyDropped;
```

Comment:
How come you didn't choose std::bitset? It could simplify SetWeaponAlreadyDropped and WasWeaponAlreadyDropped.

Reply:
Because the implementation of bitset is up to the compiler authors. I cannot guarantee that a bitset of 560 bits has a size of exactly 70 bytes. Usually the size gets aligned to the CPU's natural word size; GCC on x64 for example has sizeof(std::bitset<560>) == 72.

Also, a save produced by an executable build with MSVC might not be compatible with a GCC or a Clang executable.

Reply:
Ouch, that's quite bad for a standard. One could cmake around it, but bleargh and why bother.